src/io.rs

```diff
@@ -15,14 +15,14 @@ impl From<DecodeError> for Error {
 
 /// Prefix varint code a value and write it to `w`.
 #[inline]
-pub fn write_prefix_varint<PV: PrefixVarInt, W: Write>(v: PV, w: &mut W) -> Result<()> {
+pub fn write_prefix_varint<PV: PrefixVarInt>(v: PV, w: &mut impl Write) -> Result<()> {
```

Comment:
I'm aware of this syntax but it obviously does not always occur to me to use it. Seems like an improvement here.